qrenderdoc/Windows/MainWindow.cpp

```diff
@@ -1408,8 +1408,9 @@ void MainWindow::on_action_Attach_to_Running_Instance_triggered()
 
 void MainWindow::on_action_Manage_Remote_Servers_triggered()
 {
-  // the manager deletes itself when all lookups terminate
-  RDDialog::show(new RemoteManager(m_Ctx, this));
+  RemoteManager *dialog = new RemoteManager(m_Ctx, this);
+  RDDialog::show(dialog);
+  delete dialog;
```

Comment:
That's a shame, I guess it gets deleted within the little dialog loop, I thought it wouldn't be processed until the main event loop further up the stack but I'm not too familiar with how Qt does all of its event processing.

I suppose the solution then is just a simple refcount, where the MainWindow will keep a reference until it returns from `on_action_Manage_Remote_Servers_triggered`, and each lookup will keep a reference as well, then the window gets deleted when there are no references left.